src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,250 @@
+            
+        } catch (RouteNotFoundException ex) {
+
+            LOGGER.warning(ex.getMessage());
+
+            String command = ex.getRouteCommand();
+
+            String response = "*Help:*\n";
+            if (command.split("\\s+").length > 1)
+                response += "`"+command+"` _is an unknown command, try one of the following:_\n\n";
+            else
+                response += "\n";
+
+            for (CommandRouter.Route route : router.getRoutes()) {
+                response += "`"+route.command+"`\n```"+route.commandDescription+"```";
+                response += "\n\n";
+            }
+
+            return new JsonResponse(new SlackTextMessage(response), StaplerResponse.SC_OK);
+
+        } catch (CommandRouterException ex) {
+            LOGGER.warning(ex.getMessage());
+            return new JsonResponse(new SlackTextMessage(ex.getMessage()), StaplerResponse.SC_OK);
+
+        } catch (Exception ex) {
+            LOGGER.warning(ex.getMessage());
+            return new JsonResponse(new SlackTextMessage("An error occured: "+ ex.getMessage()), StaplerResponse.SC_OK);
+        }
+    }
+
+    public SlackTextMessage getProjectLog(String projectName,
+        String buildNumber) {
+
+        ACL.impersonate(ACL.SYSTEM);
+
+        Project project =
+            Jenkins.getInstance().getItemByFullName(projectName, Project.class);
+
+        if (project == null)
+            return new SlackTextMessage("Could not find project ("+projectName+")\n");
+
+        AbstractBuild build =
+            project.getBuildByNumber(Integer.parseInt(buildNumber));
+
+        if (build == null)
+            return new SlackTextMessage("Could not find build #"+buildNumber+" for ("+projectName+")\n");
+
+        List<String> log = new ArrayList<String>();
+        try {
+            log = build.getLog(25);
+        } catch (IOException ex) {
+            return new SlackTextMessage("Error occured returning log: "+ex.getMessage());
+        }
+
+        String response = "*"+projectName+"* *#"+buildNumber+"*\n";
+        response += "```";
+        for (String line : log) {
+            response += line + "\n";
+        }
+        response += "```";
+        return new SlackTextMessage(response);
+    }
+
+    public SlackTextMessage scheduleJob(String projectName) {
+        ACL.impersonate(ACL.SYSTEM);
+        String response = "";
+
+        Project project =
+            Jenkins.getInstance().getItemByFullName(projectName, Project.class);
+
+        boolean success = false;
+
+        if (project != null)
+            success = project.scheduleBuild(new SlackWebhookCause(this.slackUser));
+        else
+            return new SlackTextMessage("Could not find project ("+projectName+")\n");
+
+        if (success) 
+            return new SlackTextMessage("Build scheduled for project "+ projectName+"\n");
+        else
+            return new SlackTextMessage("Build not scheduled due to an issue with Jenkins");
```

Comment:
Although it seems to be totally correct, this method does have quite a few decision paths, which could be confusing. How about something like this:

```java
        Project project =
            Jenkins.getInstance().getItemByFullName(projectName, Project.class);
        
        if (project == null)
        	return new SlackTextMessage("Could not find project ("+projectName+")\n");

        if (project.scheduleBuild(new SlackWebhookCause(this.slackUser))) 
            return new SlackTextMessage("Build scheduled for project "+ projectName+"\n");
        else
            return new SlackTextMessage("Build not scheduled due to an issue with Jenkins");
```